spock-specs/src/test/groovy/org/spockframework/smoke/extension/TempDirExtensionSpec.groovy

```diff
@@ -0,0 +1,113 @@
+package org.spockframework.smoke.extension
+
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Stepwise
+import spock.lang.TempDir
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+/**
+ * @author dqyuan
+ */
+@Stepwise
+class TempDirExtensionSpec extends Specification {
+
+  @Shared
+  @TempDir
+  Path sharedDir
+  @Shared
+  Path previousShared
+
+  @TempDir
+  File iterationDir
+  @Shared
+  File previousIteration
+
+  @TempDir(baseDir = "build")
+  Path customerTempDir
+
+  @TempDir(baseDir = "build", keep = { true })
+  Path reserveDir
```

Comment:
Please select these in a shared field and delete them in the end.
I think you can select them in `setup` and then delete them in `cleanupSpec`.
Then you can test they survive, but they do not pollute the build directory.